flocker/apiclient/_client.py

```diff
@@ -0,0 +1,139 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Client for the Flocker REST API.
+"""
+
+from uuid import UUID, uuid4
+
+from zope.interface import Interface, implementer
+
+from pyrsistent import PClass, field, pmap_field, pmap
+
+from twisted.internet.defer import succeed, fail
+
+
+class Dataset(PClass):
+    """
+    A dataset in the configuration.
+
+    :param UUID primary: The node where the dataset should manifest.
+    :param int maximum_size: Size of new dataset, in bytes.
+    :param dataset_id: The UUID of the dataset.
+    :param metadata: A mapping between unicode keys and values.
+    :param bool deleted: If true indicates this dataset should be deleted.
+    """
+    dataset_id = field(type=UUID)
+    primary = field(type=UUID)
+    maximum_size = field(type=int)
+    deleted = field(type=bool)
+    metadata = pmap_field(unicode, unicode)
+
+
+class DatasetState(PClass):
+    """
+    The state of a dataset in the cluster.
+
+    :param UUID primary: The node where the dataset should manifest.
+    :param int maximum_size: Size of new dataset, in bytes.
+    :param dataset_id: The UUID of the dataset.
+    """
+    dataset_id = field(type=UUID)
+    primary = field(type=UUID)
+    maximum_size = field(type=int)
+
+
+class DatasetAlreadyExists(Exception):
+    """
+    The suggested dataset ID already exists.
+    """
+
+
+class IFlockerAPIV1(Interface):
+    """
+    The Flocker REST API, v1.
```

Comment:
I feel ill at ease calling this the REST API when it is – AIUI – the interface to a couple of clients of the REST API. 

Perhaps "The Flocker REST API client, v1"?